src/components/activity-header/nav-pages.tsx

```diff
@@ -13,20 +14,35 @@ interface IProps {
 export class NavPages extends React.PureComponent <IProps> {
   render() {
     return (
-      <div className="nav-pages" data-cy="nav-pages">
-        { this.renderHomePageButton() }
-        { this.props.pages.filter((page) => !page.is_hidden).map((page, index: number) => (
-            this.renderPageButton(page, index)
-          ))
-        }
+      <div className="nav-pages" data-cy="nav-pages" >
+        {this.renderHomePageButton()}
+        <ReactPaginate
+          previousLabel={"<"}
+          nextLabel={">"}
+          breakLabel={"..."}
+          breakLinkClassName={"page-button"}
+          pageCount={this.props.pages.length}
+          onPageChange={this.handlePaginate}
+          marginPagesDisplayed={2}
+          pageRangeDisplayed={5}
+          containerClassName={"paginate-container"}
+          pageClassName={""}
```

Comment:
I'm assuming you don't need to pass in an empty string and can remove this prop from the component.